src/dmd/backend/cgelem.c

```diff
@@ -3576,10 +3576,13 @@ STATIC elem * eleq(elem *e, goal_t goal)
             e1->Eoper == OPvar &&
             (e2->Eoper == OPpair || e2->Eoper == OPrpair) &&
             goal == GOALnone &&
-            !el_appears(e2, e1->EV.sp.Vsym)
+            !el_appears(e2, e1->EV.sp.Vsym) &&
+            // Disable this rewrite if we're using x87 and the OPpair isn't made
+            // of FP values
+            (config.fpxmmregs || tyfloating(e2->E1->Ety) == tyfloating(e2->Ety))
```

Comment:
tyfloating doesn't return a bool:
```
inline unsigned tyfloating(tym_t ty) { return tytab[ty & 0xFF] & (TYFLreal | TYFLimaginary | TYFLcomplex); }
```
so the test should be `tyfloating() != 0`

Reply:
Fixed.